library/core/src/ptr/mod.rs

```diff
@@ -446,10 +450,15 @@ mod mut_ptr;
 ///   additional invariants - this is type-dependent.
 ///
 /// Additionally, if `T` is not [`Copy`], using the pointed-to value after
-/// calling `drop_in_place` can cause undefined behavior. Note that `*to_drop =
+/// calling `drop_in_place` may cause undefined behavior. Note that `*to_drop =
 /// foo` counts as a use because it will cause the value to be dropped
 /// again. [`write()`] can be used to overwrite data without causing it to be
-/// dropped.
+/// dropped. Read operations may be UB based on library invariants of that type,
+/// for example reading the value pointed to by a dropped `Box<T>` is a use-after-free.
+///
+/// Having an `&` or `&mut` reference to the pointed-to value after calling [`drop_in_place()`]
+/// is still sound as long as it is not read from (in which case the soundness of the operation
+/// depends on the specific type).
```

Comment:
I'm not so sure that we should say this. `drop_in_place` does not on its own deinitialize, but `Drop::drop` code might, and it's undecided if holding a `&` to uninit data is sound.